src/TestProblem/ELBDM/LSS/Init_TestProb_ELBDM_LSS.cpp

```diff
@@ -89,21 +94,74 @@ void SetParameter()
-   ReadPara->Add( "LSS_InitMode",      &LSS_InitMode,          1,             1,                2                 );
-
+   ReadPara->Add( "LSS_InitMode",        &LSS_InitMode,          1,             1,                2                 );
+   ReadPara->Add( "ZoomIn_MaxLvOutside", &ZoomIn_MaxLvOutside,   MAX_LEVEL,     0,                MAX_LEVEL         );
+ 
    ReadPara->Read( FileName );
 
    delete ReadPara;
-
-// (1-2) set the default values
-
-// (1-3) check the runtime parameters
-
-
-// (2) set the problem-specific derived parameters
-
-
-// (3) reset other general-purpose parameters
+     
+   if ( OPT__FLAG_REGION )
+   {
+     if ( ELBDM_FIRST_WAVE_LEVEL > ZoomIn_MaxLvOutside )
+      Aux_Error( ERROR_INFO, " it is required to set ELBDM_FIRST_WAVE_LEVEL <= ZoomIn_MaxLvOutside for zoom-in simulation !!\n" );
+
+     char *input_line = NULL;
+     size_t len = 0;
+     int n;
+     FILE *File_zoom;
+     File_zoom = fopen( "Input__ZoominBox", "r" );
+
+//   record the number of rows in Input__ZoominBox
+     ZoomIn_MaxSnap = 0;
+     while ( getline( &input_line, &len, File_zoom ) != -1 )
+       ++ZoomIn_MaxSnap;
+     ZoomIn_MaxSnap--;
+     fclose( File_zoom );
+
+//   load Zoom-in Lagrangian Box Volumne at different redshifts
+     ZoomIn_a  = new real [ZoomIn_MaxSnap];
+     ZoomIn_Lx = new real [ZoomIn_MaxSnap];
+     ZoomIn_Ly = new real [ZoomIn_MaxSnap];
+     ZoomIn_Lz = new real [ZoomIn_MaxSnap];
+     ZoomIn_Center_x = new real [ZoomIn_MaxSnap];
+     ZoomIn_Center_y = new real [ZoomIn_MaxSnap];
+     ZoomIn_Center_z = new real [ZoomIn_MaxSnap];
+
+//   skip the header
+     File_zoom = fopen( "Input__ZoominBox", "r" );
+     getline( &input_line, &len, File_zoom );
+
+     for (int s=0; s<ZoomIn_MaxSnap; s++)
+     {
+       n = getline( &input_line, &len, File_zoom );
+       if ( n <= 1 )
+        Aux_Error( ERROR_INFO, "incorrect reading at zoom-in table at line %d of the file <%s> !!\n", s+1, "Input__ZoominBox" );
+
+       sscanf( input_line, "%f%f%f%f%f%f%f",
+               &ZoomIn_a[s],
+               &ZoomIn_Lx[s], &ZoomIn_Ly[s], &ZoomIn_Lz[s],
+               &ZoomIn_Center_x[s], &ZoomIn_Center_y[s], &ZoomIn_Center_z[s] );
+       
+       if ( s < 1 ) continue;
+
+       if ( ZoomIn_a[s-1] < ZoomIn_a[s] )
+        Aux_Error( ERROR_INFO, "Scale factors are not listed in descending order in Input__ZoominBox!!\n" );
+ 
+       if ( ZoomIn_Center_x[s-1] - ZoomIn_Lx[s-1]/2 < ZoomIn_Center_x[s] - ZoomIn_Lx[s]/2 ||
+            ZoomIn_Center_y[s-1] - ZoomIn_Ly[s-1]/2 < ZoomIn_Center_y[s] - ZoomIn_Ly[s]/2 ||
+            ZoomIn_Center_z[s-1] - ZoomIn_Lz[s-1]/2 < ZoomIn_Center_z[s] - ZoomIn_Lz[s]/2 ||
+            ZoomIn_Center_x[s-1] + ZoomIn_Lx[s-1]/2 > ZoomIn_Center_x[s] + ZoomIn_Lx[s]/2 ||
+            ZoomIn_Center_y[s-1] + ZoomIn_Ly[s-1]/2 > ZoomIn_Center_y[s] + ZoomIn_Ly[s]/2 ||
+            ZoomIn_Center_z[s-1] + ZoomIn_Lz[s-1]/2 > ZoomIn_Center_z[s] + ZoomIn_Lz[s]/2 )
```

Comment:
[Style] Two spaces around `||`:
```suggestion
       if ( ZoomIn_Center_x[s-1] - ZoomIn_Lx[s-1]/2 < ZoomIn_Center_x[s] - ZoomIn_Lx[s]/2  ||
            ZoomIn_Center_y[s-1] - ZoomIn_Ly[s-1]/2 < ZoomIn_Center_y[s] - ZoomIn_Ly[s]/2  ||
            ZoomIn_Center_z[s-1] - ZoomIn_Lz[s-1]/2 < ZoomIn_Center_z[s] - ZoomIn_Lz[s]/2  ||
            ZoomIn_Center_x[s-1] + ZoomIn_Lx[s-1]/2 > ZoomIn_Center_x[s] + ZoomIn_Lx[s]/2  ||
            ZoomIn_Center_y[s-1] + ZoomIn_Ly[s-1]/2 > ZoomIn_Center_y[s] + ZoomIn_Ly[s]/2  ||
            ZoomIn_Center_z[s-1] + ZoomIn_Lz[s-1]/2 > ZoomIn_Center_z[s] + ZoomIn_Lz[s]/2 )
```